### ISSUE TITLE
Extension keeps crashing after using it for a few hours

### ISSUE THREAD

**Issue by dionysio** [WorldBrain/Research-Engine#68]
I've imported about 20k urls, then it started crashing. First it was during the import, but now it crashes straight away after the extension starts/is reloaded. I've tried restarting the browser, but it didn't help. 

I've got this version of chrome: Version 55.0.2883.87 Built on Ubuntu , running on Ubuntu 16.04 (64-bit) and here is the part of the debug log that mentions your extension - http://pastebin.com/Fs8qL0Lr. Is there other debug info I can provide?

**Comment by blackforestboi** [comment]
Before starting the debugging, let me rephrase how I understood your  your process.

1. You go to the import section
2. There it hangs itself up, when you stumble over Http Authentification urls. (when I have those urls, it usually just takes a while until the download process continues)
Did you skip or wait in order to reach the 20k urls?
3. You press stop or restart the extension (which of both?) 
4. When using again, it crashes
What happens then? Is it just not usable when you search or does it constantly reload?

Regarding the error message, I have to resort to @obsidianart for debugging, since I am a relative beginner and he probably understands quicker what the problem is.

**Comment by dionysio** [comment]
Let me rephrase. I've used it without a problem, the import worked fine for a while. The http auth dialog was annoying, but I could just cancel it and the import continued after that. Then at some point I got a chrome notification that the extension crashed. Now whenever I try to start the extension again (either by 'Reload' on extension tab, or after restarting the browser), it crashes after a few seconds. I do not even have time to look into its settings or search from the address bar.

I've noticed 'Collect errors' option in the 'Extension' tab. After clicking on 'Errors', the last error message was:

`'Uncaught TypeError: Cannot read property 'addEventListener' of null'`

with Stacktrace: js/bundle.js:46650 (anonymous function) which is [this line](https://github.com/WorldBrain/Research-Engine/blob/5598316356ff99fe6a5a242d6de524bd5beae140/src/js-browserify/download_urls.js#L24)

**Comment by blackforestboi** [comment]
Mhh, thats weird. 

My hunch that it somehow got stuck in the setting that it should restart the extension. 

Do you have the ability to set the variable `isAbortedByUser = false` via the console?

And do you have more of that error history?

I wonder if we could improve the way the download is paused, so that we would not get to the point that the extension crashes because of this function.

**Comment by blackforestboi** [comment]
mhh or another idea, 
I think from the cancelling the import, the button might be removed and not reloaded again (as it cannot read the property of 'null'). 

I don't know at this point how to reload it, except you can try to enter/load the import page quickly?
You said you don't have the time, maybe you can load the button from the console? 

Excuse me if I can't help you too technically here, since I am quite new to JS. My expertise lay elsewhere. :/
